### PR TITLE
Switch from Grunt to Webpack

### PR DESCRIPTION
**Pull request by jacobpenny**
Using [script-loader](https://github.com/webpack/script-loader) we can make the switch from Grunt to Webpack with zero code changes. We can modularize things piece by piece in future PRs. Development becomes quite a bit easier with `webpack --watch`.

Notes:
- `script-loader` isn't compatible with strict-mode so I am using an es2015 preset which excludes it.
- 42 of the 48 files changed are compiled files

Here is a good article contrasting Grunt and Webpack:
https://advancedweb.hu/2016/02/02/the-first-steps-from-grunt-to-webpack/

TL;DR
- simpler configuration
- improves development workflow
- enables much needed [modules](https://strongloop.com/strongblog/an-introduction-to-javascript-es6-modules/) on the front-end

### CODE REVIEW

File: test/run-js-linter.sh [hunk 1]
```diff
@@ -10,6 +10,3 @@ eslint jsx/
 
 # Run ESLint on all JS files (see exceptions in .eslintignore)
 eslint htdocs/js/
-
-# Other files
-eslint Gruntfile.js
```
Comment (alisterdev) [inline test/run-js-linter.sh]: Would be nice to add lint for `webpack.config.js` instead of removed `Gruntfile.js`

File: LORIS_react.README.md [hunk 2]
```diff
@@ -51,36 +51,54 @@ curl -L https://www.npmjs.com/install.sh | sudo sh
 
 ###1.2 Installing Grunt
 
-Once NodeJS and NPM are installed, you are ready to install Grunt. First, ```cd``` into the LORIS home directory in your terminal. 
+Once NodeJS and NPM are installed, you are ready to install Webpack. First, ```cd``` into the LORIS home directory in your terminal.
```
Comment (alisterdev) [inline LORIS_react.README.md]: @driusan Should this file maybe go under `docs/` instead of root directory?

**Comment by jacobpenny** [comment]
Hold off on this, looking into another method which might make it easier to transition to modules.

**Comment by jacobpenny** [comment]
Closing in favour of #2540
